vllm/v1/core/kv_cache_utils.py

```diff
@@ -1350,8 +1425,33 @@ def get_kv_cache_configs(
                     "The KV cache specs for the same layer are different "
                     "across workers. This is not supported yet."
                 )
+
+    # Get global KV cache groups. This also handles spec unification for
+    # hybrid models when disable_hybrid_kv_cache_manager is enabled.
+    # After this call, merged_kv_cache_specs may be modified in-place.
     global_kv_cache_groups = get_kv_cache_groups(vllm_config, merged_kv_cache_specs)
 
+    # Get unified per-worker specs by projecting from the merged specs.
+    # This ensures auto-fit and memory checks use the same unified specs
+    # that will be used for actual allocation.
+    unified_kv_cache_specs = [
+        {layer: merged_kv_cache_specs[layer] for layer in spec_one_worker}
+        for spec_one_worker in kv_cache_specs
+    ]
```

Comment:
We need to consider the padding inside kv cache group. For example, if we have 8 full attention layer and 9 sliding window layers, we will pad it to 9 full attention + 9 sliding window. Happy to help on fixing this if you want.